---
title: "Myosin coverage - 206 genomes with mapq3 filter"
author: "Alyssa Yoxsimer"
date: "4/9/24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# All installed under v3.6, and v4.2.2 with exception of mapproj, gdsfmt, SNPRelate
library(tidyverse); theme_set(theme_classic())
library(fs)
library(ggpubr)
library(ggbeeswarm)
#library(gdsfmt)
#library(SNPRelate)
library(ggrepel)
library(cowplot); theme_set(theme_cowplot(font_size=6))
library(ggtree)
library(ape)
library(patchwork)
#library(mapproj)
#library(sf)
#library(rnaturalearth)
```

```{r}
# File paths for gasAcu5 masked aligned reads
ga5_roi_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_roi/roi_coverage_mapq3.txt"
metadata_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/metadata_227genomes.txt"
ga5_sim_roi_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/02_coverage/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/roi_coverage_mapq3.txt"
```
```{r}
# File paths for SNP data
#vcf_file_1 <- "/labs/kingsley/ambenj/myosin/ecotypic_depth_gasAcu1-4/227_genomes.final.filtered_chrXIX_2614822-2744076.vcf"
vcf_file_1 <- "/labs/kingsley/ambenj/myosin_dups/analysis/assemblies/gasAcu1-4/227_genomes.final.filtered.MYHSensitiveEcopeak.noDup.recode.vcf"
het_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/assemblies/gasAcu1-4/227_genomes.final.filtered.MYHSensitiveEcopeak.noDup.het"
```
```{r}
# Output file paths
out_table_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/ecotypic_depth_samples.txt"
```

```{r}
# Read roi file for gasAcu5 masked aligned reads
ga5_roi <- read_tsv(ga5_roi_file) %>% 
  mutate(bam = str_remove_all(bam, ".recal.realignGA5_C4masked.sort.merged.mkdup.bam|.recal.realignGA5_C4masked.sort.mkdup.bam|.recal.realignGA5_C4masked.sort.mkdup_mapq3_roi_coverage.txt|05_mkdup_index/"),
         type = "real") %>% 
  rename(samp = bam)
ga5_roi

# Read simulations roi file for gasAcu5 masked aligned reads
ga5_sim_roi <- read_tsv(ga5_sim_roi_file) %>%
  mutate(bam = str_remove_all(bam, ".RG.sorted.bam|/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/01_alignment/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/"),
         type = "sim") %>%
  rename(samp = bam)
ga5_sim_roi

# Combine roi tables from real and sim samples
ga5_comb_roi <- rbind(ga5_roi, ga5_sim_roi)
ga5_comb_roi %>% 
  filter(type=="sim")
```



```{r, message=FALSE, show_col_types = FALSE}
# Read whole genome files for gasAcu5 masked aligned reads
ga5_wg_files <- dir_ls(path = "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_wg/", glob = "*mapq3_wg_coverage.txt")
ga5_sim_wg_files <- dir_ls(path = "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/02_coverage/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/", glob = "*mapq3_wg_coverage.txt")

# function to add file name to dataframe
read_and_record_filename <- function(filename){
  read_tsv(filename) %>%
  mutate(filename = path_file(filename))
  }

# gather real wg files into dataframe
ga5_wg <- map_df(ga5_wg_files, read_and_record_filename)%>% 
  mutate(samp = str_remove_all(filename, ".recal.realignGA5_C4masked.sort.merged.mkdup_mapq3_wg_coverage.txt|.recal.realignGA5_C4masked.sort.mkdup_mapq3_wg_coverage.txt|../ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_wg"))
ga5_wg

# Gather simualated wg files into dataframe
ga5_sim_wg <- map_df(ga5_sim_wg_files, read_and_record_filename)%>%
  mutate(samp = str_remove_all(filename, ".mapq3_wg_coverage.txt"))
ga5_sim_wg

# combine real and simulated data into one dataframe
ga5_comb_wg <- rbind(ga5_wg, ga5_sim_wg)
ga5_comb_wg
```

Add metadata
```{r}
metadata <- read_tsv(metadata_file)
metadata
```


Is the depth consistent across chromosomes?
```{r fig.height=40, fig.width = 25}
# Plot coverage by chromosome
ga5_wg %>% 
  filter(!chr %in% c("chrM", "chrUn")) %>% 
  ggplot(aes(chr, meandepth)) +
  geom_point() +
  coord_flip() +
  facet_wrap(~samp)
```
Samples that look like they are actually males: 

* BIGR_1_32_2007_02
* BK70_X_2010_02
* BNST_X_2006_10
* LAUR_X_1993_9_5
* MIDF_REND_2011_05
* MIDF_REND_2011_06
* NTRW_X_X_02
* REIV_X_2011_04

```{r fig.height=4, fig.width = 16}
# Plot coverage by chromosome for two normal samples and the samples that look like males
ga5_wg %>% 
  filter(!chr %in% c("chrM", "chrUn"), 
         samp %in% c("AKMA_X_2001_102", "AKST_X_2001_03", "BIGR_1_32_2007_02", "BK70_X_2010_02", "BNST_X_2006_10", "LAUR_X_1993_9_5", "MIDF_REND_2011_05", "MIDF_REND_2011_06", "NTRW_X_X_02", "REIV_X_2011_04")) %>% 
  ggplot(aes(chr, meandepth)) +
  geom_point() +
  coord_flip() +
  facet_wrap(~samp, nrow=1)
```

Some samples look like they have very uneven coverage...why would this be?

```{r}
# get whole genome mean depth for gasAcu5 masked aligned (excluding chrUn, chrM, chrY) per sample 
ga5_wg_cov <- ga5_comb_wg %>% 
  filter(!chr %in% c("chrUn", "chrM", "chrY")) %>% 
  group_by(samp) %>% 
  summarize(weighted_mean_depth = weighted.mean(meandepth, endpos))
ga5_wg_cov
```



```{r fig.height=15, fig.width=26}
# Add additional depth and metadata info to rois
ga5_roi_cov <- left_join(ga5_comb_roi, ga5_wg_cov) %>% 
  left_join(., metadata, by = c("samp" = "seq_ID")) %>% 
   mutate(wg_norm_depth = meandepth / weighted_mean_depth,
          ecotype = case_when(mar_fresh == "M" ~ "Marine",
                              mar_fresh == "F" ~ "Freshwater",
                              type == "sim" ~ "Simulation"),
          ecotype = as.factor(ecotype), 
         ecotype = relevel(ecotype, "Marine"),
         sex = case_when(samp %in% c("BIGR_1_32_2007_02", "BK70_X_2010_02", "BNST_X_2006_10", "LAUR_X_1993_9_5", "MIDF_REND_2011_05", "MIDF_REND_2011_06", "NTRW_X_X_02", "REIV_X_2011_04", "rabs_THREE_spine.male.fa_sim4X.rep0", "stickleback_v5-BOULxBDGB_F5-4copy.male.fa_sim4X.rep0", "stickleback_v5-BEPA-AF_ONT_M1020-5copy.male.fa_sim4X.rep0", "stickleback_v5-BLAU22_12-6copy.male.fa_sim4X.rep0") ~ "male",
                         TRUE ~ "female"),
         samp_length = str_length(samp)) %>% 
  separate(samp, into = "acronym", remove=FALSE, sep="_")

ga5_roi_cov %>% 
  filter(sex=="female") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 1) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```


```{r fig.width=20, fig.height=6}
# Plot just the male samples for gasAcu5 masked aligned
ga5_roi_cov %>% 
  filter(sex=="male") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 11) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```

```{r fig.width=20, fig.height=6}
# Plot just the simulated samples for gasAcu5 masked aligned
ga5_roi_cov %>% 
  filter(type=="sim") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth)) +
  geom_point() +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 11) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```

```{r fig.width=26, fig.height=8}
ga5_roi_cov %>% 
  filter(desc == "HTRA1A") %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.025, 0.75)) +
  ylim(0,3) +
  ggtitle("HTRA1 read depth") +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")
```
```{r}
# Get average from simulations
sim_average <- ga5_roi_cov %>% 
  filter(type=="sim", desc=="MYH3C3") %>% 
  mutate(CN = case_when(str_detect(samp, "3copy") ~ "3-copy",
                        str_detect(samp, "4copy") ~ "4-copy",
                        str_detect(samp, "5copy") ~ "5-copy",
                        str_detect(samp, "6copy") ~ "6-copy")) %>% 
  group_by(CN, sex) %>% 
  summarise(wg_norm_depth = mean(wg_norm_depth))

sim_average
```


```{r fig.width=26, fig.height=4}
# Plot MYH3C3 read depth for all samples  
ga5_roi_cov %>% 
  filter(desc == "MYH3C3", sex=="female", type=="real") %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.025, 0.75)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=20, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         is.na(`used _river_comparisons`)) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.03, 0.75)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=17, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons and the low depth original Jones et al. 2012 samples
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         is.na(`used _river_comparisons`),
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.04, 0.79)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```


```{r fig.width=4, fig.height=4}
# Plot MYH3C3 read depth for locations with multiple samples
ga5_roi_cov
multi_samp <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         samp_length > 6) %>% 
  group_by(pop) %>% 
  mutate(pop = case_when(acronym == "BOUL" ~ "Boulton Lake, Haida Gwaii, BC, Canada",
                         pop == "Drizzle Lake, Haida Gwaii" ~ "Drizzle Lake, Haida Gwaii, BC, Canada",
                         pop == "Friant River" ~ "Friant River, CA, USA",
                         acronym == "GOLD" ~ "Gold Creek, Haida Gwaii, BC, Canada",
                         acronym == "MAYR" ~ "Mayer Lake, Haida Gwaii, BC, Canada",
                         acronym == "RDSP" ~ "Roadside Pond, Haida Gwaii, BC, Canada",
                         pop == "Spence Lake, Haida Gwaii" ~ "Spence Lake, Haida Gwaii, BC, Canada",
                         pop == "Spence Lake Outlet, Haida Gwaii" ~ "Spence Lake Outlet, Haida Gwaii, BC, Canada",
                         pop == "Vifissta_avatn" ~ "Vifisstaðavatn Lake, Iceland",
                         pop == "Wallace Lake" ~ "Wallace Lake, AK, USA",
                         TRUE ~ pop)) %>% 
  mutate(pop_total= n()) %>% 
  filter(pop_total > 1, is.na(`used _river_comparisons`), acronym != "LOBG") %>% 
  mutate(pop = factor(pop, levels = c("Friant River, CA, USA", 
                                     "Boulton Lake, Haida Gwaii, BC, Canada",
                                     "Spence Lake, Haida Gwaii, BC, Canada",
                                     "Spence Lake Outlet, Haida Gwaii, BC, Canada",
                                     "Drizzle Lake, Haida Gwaii, BC, Canada",
                                     "Gold Creek, Haida Gwaii, BC, Canada",
                                     "Mayer Lake, Haida Gwaii, BC, Canada",
                                     "Roadside Pond, Haida Gwaii, BC, Canada",
                                     "Wallace Lake, AK, USA",
                                     "Vifisstaðavatn Lake, Iceland")))
within_pop_plot <- multi_samp %>% 
  ggplot(aes(pop, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(size = 1.5, cex=2.5,alpha = 0.7) +
  geom_boxplot(fill = NA, outlier.shape = NA) +
  scale_color_manual(values=c("#0072b2", "#d73027")) +
  scale_y_continuous(limits = c(0,3.2), expand = expansion(mult = c(0, 0))) +
  xlab("Population") +
  ylab("MYH3C3 normalized depth") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
    theme(legend.position = c(0.05, 0.85),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          plot.margin = margin(l = 30, r = 20))

within_pop_plot
```
```{r}
# Save figure to separate file
ggsave(
  "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/within_pop_depth_figure.pdf",
  plot = within_pop_plot,
  scale = 1,
  width = 4,
  height = 4,
  units = c("in"),
  dpi = 300,
)
```


```{r fig.width=6, fig.height=4}
# Plot MYH3C3 read depth for MAYR lake
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         samp_length > 6) %>% 
  group_by(pop) %>% 
  mutate(pop_total= n()) %>% 
  filter(pop_total >1, is.na(`used _river_comparisons`), acronym != "LOBG") %>% 
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#0072b2", "#d73027")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.04, 0.79)) +
  facet_grid(cols = vars(pop), scales = "free_x", space="free") +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")

```
```{r fig.width=3, fig.height=4}
# Plot MYH3C3 read depth for MAYR lake
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         acronym=="MAYR",
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#0072b2")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.04, 0.79)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=10, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons and the low depth original Jones et al. 2012 samples
C3_global <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         c155_global_FvsM %in% c(0,1),
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#0072b2", "black")) +
   theme_cowplot(6) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.07, 0.79)) +

  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
C3_global
```


```{r}
ga5_roi_cov %>% 
  filter(desc == "MYH3C3", type=="real", sex=="female", c155_global_FvsM %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#0072b2"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude (°N)",
   ylab = "Normalized depth of MYH3C3\n(1x genome coverage)") + 
  stat_cor(aes(color = ecotype), label.x = 30) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
```


```{r}
Pacific_PNW_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", PNW_independent_MvsF_c150 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#0072b2"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3") + 
  stat_cor(aes(color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
Pacific_PNW_lat
```

```{r}
# Plot without original 21 genomes and including California
Pacific_CA_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", PNW_independent_MvsF_c150 %in% c(0,1) | CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#0072b2"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3") + 
  stat_cor(aes(color = ecotype))+
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")

Pacific_CA_lat
```

```{r}
# Plot without original 21 genomes and for populations glaciated during the last ice age
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", c154_globalsuperglacial_FvsM %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#0072b2"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3\n(1x genome coverage)") + 
  stat_cor(aes(color = ecotype)) +
      geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
```


```{r}
# Plot without original 21 samples
Europe_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#0072b2"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "MYH3C3 normalized depth") + 
  stat_cor(aes(color = ecotype)) +
      geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
Europe_lat
```



```{r}
# Plot without original 21 samples
test <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", c155_global_FvsM %in% c(0,1), samp_length >6) %>%
  ggplot(aes(ecotype, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(size = 3, cex=2.5,alpha = 0.7) +
  geom_boxplot(fill = NA) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
  xlab("Ecotype") +
  ylab("MYH3C3 normalized depth") +
  theme_cowplot(6) +
    theme(legend.position = "top") +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 2.6, size = 3)
test
```

```{r fig.width=5, fig.height=5}
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", str_detect(samp, "LOBG")) %>%
  ggplot(aes(pop, wg_norm_depth, color = pop)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(size = 3, cex=2.5,alpha = 0.7) +
  geom_boxplot(fill = NA) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
  ggtitle("MYH3C3 read depth") +
  xlab("Ecotype") +
  ylab("Normalized depth of MYH3C3 \n(1x genome coverage)") +
  theme(legend.position = "none",
        text = element_text(size=16)) +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 2.6, size = 5)
```




```{r}
# Make wider version of table to plot correlation between different genes
ga5_roi_cov_wide <- ga5_roi_cov %>% 
  select(desc, samp:type, weighted_mean_depth:sex) %>% 
  pivot_wider(names_from = desc, values_from = wg_norm_depth)
```

```{r}
# Plot pearson correlation between MYH3C3 read depth and duplication region
ga5_roi_cov_wide %>% 
  filter(sex=="female", type=="real") %>%
  ggscatter(x = "MYH3C3", y = "duplication_region",
   add = "reg.line", conf.int = TRUE, alpha=0.5) +
  stat_cor()
```


## Final figure plots

```{r}
# Plot MYH3C3 read depth for balanced global selection, excluding low depth original Jones et al. 2012 samples
ga5_roi_cov_filt <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1)|PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1), samp_length > 6) %>%
  mutate(Individual = case_when(samp == "BIGR_1_32_2007_03" ~ "BIGR_M",
                                samp == "BIGR_52_54_2008_02" ~ "BIGR_F",
                                samp == "LITC_0_05_2008_841" ~ "LITC_M",
                                samp == "LITC_23_32_2008_306" ~ "LITC_F",
                                samp == "MIDF_BLUP_2011_01" ~ "MIDF_M",
                                samp == "MIDF_REND_2011_01" ~ "MIDF_F",
                                samp == "TYNE_1_2001_14" ~ "TYNE_M",
                                samp == "TYNE_8_2003_902" ~ "TYNE_F",
                                TRUE ~ acronym))
# Wilcoxon test for marine vs freshwater
#w <- wilcox.test(subset(ga5_roi_cov_filt, ecotype == "Freshwater")$wg_norm_depth, subset(ga5_roi_cov_filt, ecotype == "Marine")$wg_norm_depth, alternative = "greater")
w <- wilcox.test(subset(ga5_roi_cov_filt, ecotype == "Freshwater")$wg_norm_depth, subset(ga5_roi_cov_filt, ecotype == "Marine")$wg_norm_depth, alternative = "two.sided")
w

C3_global <- ga5_roi_cov_filt %>% 
  ggplot(aes(fct_reorder(Individual, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey", linewidth=0.25) +
  geom_segment( aes(x=Individual, xend=Individual, y=0, yend=wg_norm_depth)) +
  geom_point(size=1.5) +
  scale_color_manual(values=c("#d73027","#0072b2"), name = "Ecotype") +
#  scale_shape_manual(values = c(21,19)) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
   theme_cowplot(6) +
    panel_border(color="black", size=0.75) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), 
        legend.position = c(0.02, 0.8),
        legend.box.background = element_rect(color="black", linewidth=0.25, fill = "white"),
      legend.box.margin = margin(4, 4, 4, 4)) +
  xlab("Individual") +
  ylab("MYH3C3 normalized depth")
#    ylim(0,3.5)
C3_global

```


```{r}
# Plot with facet
lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1)|PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1), samp_length > 6) %>%
  mutate(ocean = case_when(PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1) ~ "Pacific",
                           NorthEurope_independent_MvsF_c151 %in% c(0,1) ~ "Atlantic")) %>% 
  ggplot(aes(GPS_north, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey", linewidth=0.25) +
  geom_point(alpha=0.7, size=1.5) +
  geom_smooth(aes(fill = ecotype), method = "lm") +
  stat_cor(aes(color = ecotype), size=2, method = "pearson", cor.coef.name = "R") +
 # stat_cor(aes(color = ecotype), size=2, method = "spearman", cor.coef.name = "rho") +
  scale_color_manual(values=c("#d73027","#0072b2"),) +
  scale_fill_manual(values=c("#d73027","#0072b2")) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
  facet_wrap(~factor(ocean, levels=c("Pacific", "Atlantic")), scales = "free_x") +
  theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  xlab("Latitude (°N)") +
  ylab("MYH3C3 normalized depth")
  
lat
```

```{r}
hz_plot <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3", sex == "female", !is.na(`used _river_comparisons`), samp_length > 6) %>% 
  mutate(location = as.factor(case_when(acronym == "BIGR" ~ "Big River,\nCA, USA",
                              acronym %in% c("BNMA", "BNST") ~ "Bonsall River,\nBC, Canada",
                              acronym == "LITC" ~ "Little Campbell River,\nBC, Canada",
                              acronym == "TYNE" ~ "Tyne River,\nScotland, UK",
                              acronym == "MIDF" ~ "Midfjardara River,\nIceland")),
         location = fct_relevel(location, "Big River,\nCA, USA", "Little Campbell River,\nBC, Canada", "Bonsall River,\nBC, Canada", "Tyne River,\nScotland, UK", "Midfjardara River,\nIceland")) %>% 
  ggplot(aes(location, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey", linewidth=0.25) +
  geom_beeswarm(cex=2, dodge.width = .7, alpha = 0.7) +
  geom_boxplot(fill = NA, outlier.shape = NA) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
#  ylim(0,2.5) +
  ylab("MYH3C3 normalized depth") +
  theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none",
        axis.title.x=element_blank()) +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 3.3, size = 3, label="p.signif")
hz_plot
```



```{r}
# Arrange plots using cowplot
# align all plots vertically
plots <- align_plots(test, C3_global, lat, hz_plot, align = 'v', axis = 'l')

top_row <- plot_grid(
  plots[[2]],
  labels = c("B"),
  label_size = 12,
  nrow = 1
)

bottom_row <- plot_grid(
  plots[[3]], plots[[4]],
  labels = c("C", "D"),
  rel_widths = c(1, 1),
  label_size = 12,
  nrow = 1
)
final_plot <- plot_grid(top_row, bottom_row, ncol = 1, rel_heights = c(1.1, 1), label_size = 12)
final_plot
```

```{r}
# Save figure to separate file
ggsave(
  "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/ecotypic_depth_figure.pdf",
  plot = final_plot,
  scale = 1,
  width = 6.5,
  height = 3.75,
  units = c("in"),
  dpi = 300,
)
```

```{r}
# Make table of samples used for figure
fig_table <- ga5_roi_cov %>% 
#  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1)|PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1)|!is.na(`used _river_comparisons`), samp_length > 6) %>%
  filter(desc == "MYH3C3",  type=="real", sex=="female", samp_length > 6) %>%
  mutate(global_analysis = case_when(NorthEurope_independent_MvsF_c151 %in% c(0,1)|PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1) ~ "yes",
                                     TRUE ~ "no")) %>% 
  mutate(Figure1B_label = case_when(samp == "BIGR_1_32_2007_03" ~ "BIGR_M",
                                samp == "BIGR_52_54_2008_02" ~ "BIGR_F",
                                samp == "LITC_0_05_2008_841" ~ "LITC_M",
                                samp == "LITC_23_32_2008_306" ~ "LITC_F",
                                samp == "MIDF_BLUP_2011_01" ~ "MIDF_M",
                                samp == "MIDF_REND_2011_01" ~ "MIDF_F",
                                samp == "TYNE_1_2001_14" ~ "TYNE_M",
                                samp == "TYNE_8_2003_902" ~ "TYNE_F",
                                samp == "BNMA_X_2006_01" ~ "BNMA",
                                samp == "BNST_X2006_08" ~ "BNST",
                                !is.na(`used _river_comparisons`) ~ NA,
                                global_analysis == "yes" ~ acronym),
         analysis_group = case_when(global_analysis == "yes" & !is.na(`used _river_comparisons`) ~ "Global comparison, Hybrid zone",
                                    global_analysis == "yes" & samp %in% multi_samp$samp ~ "Global comparison, Within population",
                                    global_analysis == "yes" ~ "Global comparison",
                                    samp %in% multi_samp$samp ~ "Within population",
                                    !is.na(`used _river_comparisons`) ~ "Hybrid zone"),
         region = case_when(PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1) ~ "Pacific",
                           NorthEurope_independent_MvsF_c151 %in% c(0,1) ~ "Atlantic",
                           str_detect(samp, "MIDF|TYNE") ~ "Atlantic",
                           str_detect(samp, "BNMA|BNST|BIGR|LITC") ~ "Pacific"),
         wg_norm_depth = round(wg_norm_depth, digits = 3)) %>% 
  select(samp, Figure1B_label, pop, GPS_north, GPS_east, ecotype, water_type, wg_norm_depth, analysis_group, region, NorthEurope_independent_MvsF_c151, PNW_independent_MvsF_c150, CaliforniaFreshwater_vs_AllPacificMarine_c153)

# Write table to output file
colnames(fig_table) <- c("Sample", "Figure1B label", "Population", "GPS (North)", "GPS (East)", "Ecotype", "Water type", "MYH3C3 normalized depth", "Analysis group(s)", "Ocean basin", "NorthEurope_independent_MvsF_c151", "PNW_independent_MvsF_c150", "CaliforniaFreshwater_vs_AllPacificMarine_c153")
fig_table
fig_table %>% 
  write_tsv(out_table_file)
```

```{r}
# Count n in each group for global analysis
fig_table %>% 
  filter(str_detect(`Analysis group(s)`, "Global")) %>% 
  group_by(`Ocean basin`, Ecotype) %>% 
  summarize(n = n())
```
```{r}
# Count n in each group for global analysis
fig_table %>% 
  filter(str_detect(`Analysis group(s)`, "Within")) %>% 
  group_by(Population) %>% 
  summarize(n = n())
```


```{r}
# Count n in each group for hybrid zone analysis
fig_table %>% 
  filter(str_detect(`Analysis group(s)`, "Hybrid")) %>% 
  group_by(Population, Ecotype) %>% 
  summarize(n = n())
```

```{r}
fig_table %>% 
  filter(str_detect(`Analysis group(s)`, "Global")) %>% 
  group_by()
```

## Plot phylogenetic tree with read depth
```{r fig.height=8, fig.width=5}
# Read tree
tree <- read.tree("/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/phylogeny/subsample.eco.min4.phy.treefile")


# Extract names from tree and standardize names for join
metadata <- data.frame(tip=tree$tip.label) %>% 
  mutate(Sample = str_replace_all(tip, "\\|", "_")) %>% 
  left_join(., fig_table) 

p_tree <- ggtree(tree) %<+% metadata +
  geom_tiplab(aes(label= `Figure1B label`, color = Ecotype), size = 2, align=TRUE, linesize=.5) +
  geom_text2(aes(
      subset = !isTip & as.numeric(label) >= 70, label = label), hjust = 1.1, vjust = 1.2, size = 1.8) +
#  geom_text2(aes(label = node), hjust = 1.1, vjust = 1.2, size = 2) +
  scale_size(range = c(1, 5)) +
  geom_treescale(x = 0.02, y = 95, width = 0.05, fontsize = 2) +
  scale_color_manual(values = c(Freshwater="#0072b2", Marine="#d73027")) +
  theme_tree() +
  theme(legend.position = "none") +
  xlim_tree(0.35) +
  coord_cartesian(clip = "off")


# Extract tip order from the tree
tip_order <- p_tree$data %>%
  dplyr::filter(isTip) %>%
  dplyr::arrange(y) %>%
  dplyr::pull(label)

metadata$tip <- factor(metadata$tip, levels = tip_order)

# Make a lollipop plot using ggplot2
p_lollipop <- ggplot(metadata, aes(y = tip, color = Ecotype)) +
  geom_vline(data = sim_average, aes(xintercept = wg_norm_depth), linetype = "dashed", color = "grey", linewidth=0.25) +
  geom_segment(aes(x = 0, xend = `MYH3C3 normalized depth`), linewidth = 0.6) +
  geom_point(
    aes(x = `MYH3C3 normalized depth`),
    size = 1.5
  ) +
  scale_color_manual(values = c(Freshwater="#0072b2", Marine="#d73027")) +
  scale_x_continuous(limits = c(0,4), breaks = c(0, 2, 4), expand = expansion(mult = c(0, 0))) +
  xlab("MYH3C3 normalized depth") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  theme(
    axis.title.y = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position = "top"
  )

# Combine tree and lollipop plot
p_final <- p_tree + plot_spacer() + p_lollipop +
  plot_layout(widths = c(1.5, -0.06, 0.5))

p_final
```
```{r}
# Save figure to separate file
ggsave(
  "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/subsample.eco.min4.phy_depth_plot.pdf",
  plot = p_final,
  scale = 1,
  width = 5,
  height = 8,
  units = c("in"),
  dpi = 300,
)
```


## PCA of myosin region
```{r}

# Reformat (biallelic version, can also try "copy.num.of.ref")
snpgdsVCF2GDS(vcf_file_1, "all.gds", method="biallelic.only")

# Summary
snpgdsSummary("all.gds")

# Open GDS file
genofile <- snpgdsOpen("all.gds")

genofile

# Run PCA
pca <- snpgdsPCA(genofile, num.thread=2, autosome.only = FALSE)
#pca_filter <- snpgdsPCA(genofile, num.thread=2, autosome.only = FALSE, sample.id = vcf_samples_filt)

snpgdsClose(genofile)
```

```{r}
# variance proportion (%)
pc.percent <- round(pca$varprop*100, 2)
PC1_lab <- paste("PC1 (", pc.percent[1], "%)", sep = "")
PC2_lab <- paste("PC2 (", pc.percent[2], "%)", sep = "")
```

```{r}
# make a data.frame
tab <- data.frame(sample.id = pca$sample.id,
    EV1 = pca$eigenvect[,1],    # the first eigenvector
    EV2 = pca$eigenvect[,2],    # the second eigenvector
    EV3 = pca$eigenvect[,3],    # the third eigenvector
    EV4 = pca$eigenvect[,4],    # the fourth eigenvector
    stringsAsFactors = FALSE)

tab_annot <- tab %>%
  mutate(samp = str_replace_all(sample.id, '\\|', '_'),
         samp = str_replace_all(samp, '\\#', '_')) %>%
  left_join(., filter(ga5_roi_cov, desc == "MYH3C3"), by=c("samp" = "samp")) %>% 
  left_join(., read_tsv(het_file), by=c("sample.id" = "INDV")) %>% 
  mutate(prop_hom = `O(HOM)`/N_SITES,
         prop_het = 1-prop_hom,
         het_status = ifelse(F > 0.5, "likely-hom", "likely-het"))
tab_annot

read_tsv(het_file)

# Get subset of sample names that excludes Jones et al. 2012 samples and males

vcf_samples_filt <- filter(tab_annot, sex=="female", samp_length > 6, desc=="MYH3C3", type=="real")$sample.id

```


```{r}
# make filtered version of PCA with males and Jones et al. 2012 samples
# Open GDS file
genofile <- snpgdsOpen("all.gds")

# Run PCA
pca_filter <- snpgdsPCA(genofile, num.thread=2, autosome.only = FALSE, sample.id = vcf_samples_filt)

snpgdsClose(genofile)

# variance proportion (%)
pc.percent_filter <- round(pca_filter$varprop*100, 2)
PC1_lab_filter <- paste("PC1 (", pc.percent_filter[1], "%)", sep = "")
PC2_lab_filter <- paste("PC2 (", pc.percent_filter[2], "%)", sep = "")

# make a new data.frame
tab_filter <- data.frame(sample.id = pca_filter$sample.id,
    EV1_filter = pca_filter$eigenvect[,1],    # the first eigenvector
    EV2_filter = pca_filter$eigenvect[,2],    # the second eigenvector
    EV3_filter = pca_filter$eigenvect[,3],    # the third eigenvector
    EV4_filter = pca_filter$eigenvect[,4],    # the fourth eigenvector
    stringsAsFactors = FALSE)

# Add to table
tab_annot <- left_join(tab_annot, tab_filter)
```



```{r}
# Make plot of all PCA without filtering where shape is het status
xlab
tab_annot %>%
  ggplot(aes(EV1, EV2, color = wg_norm_depth)) +
  geom_point(aes(shape = het_status), size = 3,) +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab) +
  ylab(PC2_lab)
```
```{r fig.width=20, fig.height=15}
# Make plot of all PCA without filtering where shape is het status with labels
tab_annot %>%
  ggplot(aes(EV1, EV2, color = wg_norm_depth)) +
  geom_point(aes(shape = het_status), size = 3,) +
  geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 50, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab) +
  ylab(PC2_lab)
  
```


```{r}
# Make plot of all PCA without filtering where shape is marine or freshwater
tab_annot %>%
  ggplot(aes(EV1, EV2, color = wg_norm_depth)) +
  geom_point(aes(shape = ecotype), size = 3,) +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab) +
  ylab(PC2_lab)
```

```{r fig.width=20, fig.height=15}
# Make plot of all PCA without filtering where shape is marine or freshwater with labels
tab_annot %>%
  ggplot(aes(EV1, EV2, color = wg_norm_depth)) +
  geom_point(aes(shape = ecotype), size = 3,) +
  scale_color_viridis_c(option = "magma") +
    theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
  geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 50, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
  xlab(PC1_lab) +
  ylab(PC2_lab)
```

Remake plots with filtering out males and Jones et al. 2012 samples before running the PCA

```{r}
# Make plot of filtered PCA where shape is het status
tab_annot %>%
  ggplot(aes(EV1_filter, EV2_filter, color = wg_norm_depth)) +
  geom_point(aes(shape = het_status), size = 3,) +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab_filter) +
  ylab(PC2_lab_filter)
```
```{r fig.width=20, fig.height=15}
# Make plot of filtered PCA where shape is het status with labels
tab_annot %>%
  ggplot(aes(EV1_filter, EV2_filter, color = wg_norm_depth)) +
  geom_point(aes(shape = het_status), size = 3,) +
  geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 50, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab_filter) +
  ylab(PC2_lab_filter)
  
```


```{r}
# Make plot of filtered PCA where shape is marine or freshwater
tab_annot %>%
  ggplot(aes(EV1_filter, EV2_filter, color = wg_norm_depth)) +
  geom_point(aes(shape = ecotype), size = 3,) +
  scale_color_viridis_c(option = "magma") +
  theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  xlab(PC1_lab_filter) +
  ylab(PC2_lab_filter)
```

```{r fig.width=20, fig.height=15}
# Make plot of filtered PCA where shape is marine or freshwater with labels
tab_annot %>%
  ggplot(aes(EV1_filter, EV2_filter, color = wg_norm_depth)) +
  geom_point(aes(shape = ecotype), size = 3,) +
  scale_color_viridis_c(option = "magma") +
    theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
  geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 50, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
  xlab(PC1_lab_filter) +
  ylab(PC2_lab_filter)
```


```{r fig.width=20, fig.height=15}
# Plot relationship between EV1 and read depth
tab_annot %>%
  ggplot(aes(-EV1_filter, wg_norm_depth, color=ecotype, shape=het_status)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey", linewidth=0.25) +
  geom_point(alpha=0.5, size=3) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
    geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 20, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
      theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
    xlab(paste("-",PC1_lab_filter, sep="")) +
  ylab("MYH3C3 normalized depth")
           
```

```{r fig.width=20, fig.height=15}
# Plot relationship between EV1 and heterozygosity
tab_annot %>%
  ggplot(aes(-EV1_filter, prop_het, color=ecotype, shape=het_status)) +
  geom_point(alpha=0.5, size=3) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
    geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 20, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
      theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
    xlab(paste("-",PC1_lab_filter, sep="")) +
  ylab("Proportion heterozygous sites")
           
```
```{r fig.width=20, fig.height=15}
# Plot relationship between EV1 and F (inbreeding coefficient)
tab_annot %>%
  ggplot(aes(-EV1_filter, F, color=ecotype, shape=het_status)) +
  geom_point(alpha=0.5, size=3) +
  scale_color_manual(values=c("#d73027","#0072b2")) +
    geom_label_repel(aes(label = acronym),
                   size = 3,
                  box.padding   = 0.5,
                  point.padding = 0.5,
                  max.overlaps = 20, 
                  nudge_x = 0.001,
                  segment.color = 'grey50') +
      theme_cowplot(20) +
  panel_border(color="black", size=0.75) +
    xlab(paste("-",PC1_lab_filter, sep="")) +
  ylab("F (Inbreeding)")
           
```

```{r}
# Get list of candidate California homoyzgotes
tab_annot %>% 
  filter(EV2_filter < -0.025, het_status == "likely-hom")
```

```{r}
# Get list of candidate California homoyzgotes that are the closest to the bay area
tab_annot %>% 
  filter(EV2_filter < -0.025, het_status == "likely-hom", GPS_north < 38.5, GPS_north>37)
```

```{r}
# Get list of candidate reciprocal deletion alleles (freshwater haplotype but 3 copy locus)
tab_annot %>% 
  filter(EV1_filter < -0.025, het_status == "likely-hom", wg_norm_depth < 1)
```

