---
title: "Myosin coverage - 206 genomes with mapq3 filter"
author: "Alyssa Yoxsimer"
date: "4/9/24"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
# All installed under v3.6, and v4.2.2 with exception of mapproj, gdsfmt, SNPRelate
library(tidyverse); theme_set(theme_classic())
library(fs)
library(ggpubr)
library(ggbeeswarm)
#library(mapproj)
library(gdsfmt)
library(SNPRelate)
library(ggrepel)
library(cowplot); theme_set(theme_cowplot(font_size=6))
```

```{r}
# File paths for gasAcu5 masked aligned reads
ga5_roi_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_roi/roi_coverage_mapq3.txt"
metadata_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/metadata_227genomes.txt"
ga5_sim_roi_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/02_coverage/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/roi_coverage_mapq3.txt"
```

```{r}
# Read roi file for gasAcu5 masked aligned reads
ga5_roi <- read_tsv(ga5_roi_file) %>% 
  mutate(bam = str_remove_all(bam, ".recal.realignGA5_C4masked.sort.merged.mkdup.bam|.recal.realignGA5_C4masked.sort.mkdup.bam|.recal.realignGA5_C4masked.sort.mkdup_mapq3_roi_coverage.txt|05_mkdup_index/"),
         type = "real") %>% 
  rename(samp = bam)
ga5_roi

# Read simulations roi file for gasAcu5 masked aligned reads
ga5_sim_roi <- read_tsv(ga5_sim_roi_file) %>%
  mutate(bam = str_remove_all(bam, ".RG.sorted.bam|/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/01_alignment/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/"),
         type = "sim") %>%
  rename(samp = bam)
ga5_sim_roi

# Combine roi tables from real and sim samples
ga5_comb_roi <- rbind(ga5_roi, ga5_sim_roi)
ga5_comb_roi %>% 
  filter(type=="sim")
```



```{r, message=FALSE, show_col_types = FALSE}
# Read whole genome files for gasAcu5 masked aligned reads
ga5_wg_files <- dir_ls(path = "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_wg/", glob = "*mapq3_wg_coverage.txt")
ga5_sim_wg_files <- dir_ls(path = "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/depth_simulations/206_genomes/02_coverage/stickleback_v5_assembly_MYH3C4dup_hardmasked_noChrY/", glob = "*mapq3_wg_coverage.txt")

# function to add file name to dataframe
read_and_record_filename <- function(filename){
  read_tsv(filename) %>%
  mutate(filename = path_file(filename))
  }

# gather real wg files into dataframe
ga5_wg <- map_df(ga5_wg_files, read_and_record_filename)%>% 
  mutate(samp = str_remove_all(filename, ".recal.realignGA5_C4masked.sort.merged.mkdup_mapq3_wg_coverage.txt|.recal.realignGA5_C4masked.sort.mkdup_mapq3_wg_coverage.txt|../ecotypic_depth/gasAcuv5_C4masked_nochrY/06_samtools_coverage_wg"))
ga5_wg

# Gather simualated wg files into dataframe
ga5_sim_wg <- map_df(ga5_sim_wg_files, read_and_record_filename)%>%
  mutate(samp = str_remove_all(filename, ".mapq3_wg_coverage.txt"))
ga5_sim_wg

# combine real and simulated data into one dataframe
ga5_comb_wg <- rbind(ga5_wg, ga5_sim_wg)
ga5_comb_wg
```

Add metadata
```{r}
metadata <- read_tsv(metadata_file)
metadata
```


Is the depth consistent across chromosomes?
```{r fig.height=40, fig.width = 25}
# Plot coverage by chromosome
ga5_wg %>% 
  filter(!chr %in% c("chrM", "chrUn")) %>% 
  ggplot(aes(chr, meandepth)) +
  geom_point() +
  coord_flip() +
  facet_wrap(~samp)
```
Samples that look like they are actually males: 

* BIGR_1_32_2007_02
* BK70_X_2010_02
* BNST_X_2006_10
* LAUR_X_1993_9_5
* MIDF_REND_2011_05
* MIDF_REND_2011_06
* NTRW_X_X_02
* REIV_X_2011_04

```{r fig.height=4, fig.width = 16}
# Plot coverage by chromosome for two normal samples and the samples that look like males
ga5_wg %>% 
  filter(!chr %in% c("chrM", "chrUn"), 
         samp %in% c("AKMA_X_2001_102", "AKST_X_2001_03", "BIGR_1_32_2007_02", "BK70_X_2010_02", "BNST_X_2006_10", "LAUR_X_1993_9_5", "MIDF_REND_2011_05", "MIDF_REND_2011_06", "NTRW_X_X_02", "REIV_X_2011_04")) %>% 
  ggplot(aes(chr, meandepth)) +
  geom_point() +
  coord_flip() +
  facet_wrap(~samp, nrow=1)
```

Some samples look like they have very uneven coverage...why would this be?

```{r}
# get whole genome mean depth for gasAcu5 masked aligned (excluding chrUn, chrM, chrY) per sample 
ga5_wg_cov <- ga5_comb_wg %>% 
  filter(!chr %in% c("chrUn", "chrM", "chrY")) %>% 
  group_by(samp) %>% 
  summarize(weighted_mean_depth = weighted.mean(meandepth, endpos))
ga5_wg_cov
```



```{r fig.height=15, fig.width=26}
# Add additional depth and metadata info to rois
ga5_roi_cov <- left_join(ga5_comb_roi, ga5_wg_cov) %>% 
  left_join(., metadata, by = c("samp" = "seq_ID")) %>% 
   mutate(wg_norm_depth = meandepth / weighted_mean_depth,
          ecotype = case_when(mar_fresh == "M" ~ "Marine",
                              mar_fresh == "F" ~ "Freshwater",
                              type == "sim" ~ "Simulation"),
          ecotype = as.factor(ecotype), 
         ecotype = relevel(ecotype, "Marine"),
         sex = case_when(samp %in% c("BIGR_1_32_2007_02", "BK70_X_2010_02", "BNST_X_2006_10", "LAUR_X_1993_9_5", "MIDF_REND_2011_05", "MIDF_REND_2011_06", "NTRW_X_X_02", "REIV_X_2011_04", "rabs_THREE_spine.male.fa_sim4X.rep0", "stickleback_v5-BOULxBDGB_F5-4copy.male.fa_sim4X.rep0", "stickleback_v5-BEPA-AF_ONT_M1020-5copy.male.fa_sim4X.rep0", "stickleback_v5-BLAU22_12-6copy.male.fa_sim4X.rep0") ~ "male",
                         TRUE ~ "female"),
         samp_length = str_length(samp)) %>% 
  separate(samp, into = "acronym", remove=FALSE, sep="_")

ga5_roi_cov %>% 
  filter(sex=="female") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 1) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```


```{r fig.width=20, fig.height=6}
# Plot just the male samples for gasAcu5 masked aligned
ga5_roi_cov %>% 
  filter(sex=="male") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 11) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```

```{r fig.width=20, fig.height=6}
# Plot just the simulated samples for gasAcu5 masked aligned
ga5_roi_cov %>% 
  filter(type=="sim") %>% 
#  mutate(desc = fct_relevel(desc, "eda", "nlrc5", "calb2a")) %>% 
  ggplot(aes(reorder(samp, wg_norm_depth), wg_norm_depth)) +
  geom_point() +
  theme(axis.text.x=element_text(angle = -90, hjust = 0)) +
  facet_wrap(~desc, ncol = 11) +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")

```

```{r fig.width=26, fig.height=8}
ga5_roi_cov %>% 
  filter(desc == "HTRA1A") %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.025, 0.75)) +
  ylim(0,3) +
  ggtitle("HTRA1 read depth") +
  xlab("Individual") +
  ylab("Normalized depth (1x genome coverage)")
```
```{r}
# Get average from simulations
sim_average <- ga5_roi_cov %>% 
  filter(type=="sim", desc=="MYH3C3") %>% 
  mutate(CN = case_when(str_detect(samp, "3copy") ~ "3-copy",
                        str_detect(samp, "4copy") ~ "4-copy",
                        str_detect(samp, "5copy") ~ "5-copy",
                        str_detect(samp, "6copy") ~ "6-copy")) %>% 
  group_by(CN, sex) %>% 
  summarise(wg_norm_depth = mean(wg_norm_depth))

sim_average
```


```{r fig.width=26, fig.height=4}
# Plot MYH3C3 read depth for all samples  
ga5_roi_cov %>% 
  filter(desc == "MYH3C3", sex=="female", type=="real") %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.025, 0.75)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=20, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         is.na(`used _river_comparisons`)) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.03, 0.75)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=17, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons and the low depth original Jones et al. 2012 samples
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         is.na(`used _river_comparisons`),
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.04, 0.79)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
```

```{r fig.width=10, fig.height=4}
# Plot MYH3C3 read depth for all samples except those used in river comparisons and the low depth original Jones et al. 2012 samples
C3_global <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         c155_global_FvsM %in% c(0,1),
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point() +
  scale_color_manual(values=c("#d73027","#4575b4", "black")) +
   theme_cowplot(6) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.07, 0.79)) +

  xlab("Individual") +
  ylab("Normalized MYH3C3 read depth")
C3_global
```


```{r}
ga5_roi_cov %>% 
  filter(desc == "MYH3C3", type=="real", sex=="female", c155_global_FvsM %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#4575b4"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude (°N)",
   ylab = "Normalized depth of MYH3C3\n(1x genome coverage)") + 
  stat_cor(aes(color = ecotype), label.x = 30) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
```


```{r}
Pacific_PNW_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", PNW_independent_MvsF_c150 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#4575b4"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3") + 
  stat_cor(aes(color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
Pacific_PNW_lat
```

```{r}
# Plot without original 21 genomes and including California
Pacific_CA_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", PNW_independent_MvsF_c150 %in% c(0,1) | CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#4575b4"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3") + 
  stat_cor(aes(color = ecotype))+
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")

Pacific_CA_lat
```

```{r}
# Plot without original 21 genomes and for populations glaciated during the last ice age
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", c154_globalsuperglacial_FvsM %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#4575b4"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "Normalized depth of MYH3C3\n(1x genome coverage)") + 
  stat_cor(aes(color = ecotype)) +
      geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
```


```{r}
# Plot without original 21 samples
Europe_lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1), samp_length > 6) %>%
  ggscatter(x = "GPS_north", y = "wg_norm_depth",
   color = "ecotype", palette = c("#d73027","#4575b4"),
   add = "reg.line", conf.int = TRUE,
   xlab = "Latitude",
   ylab = "MYH3C3 normalized depth") + 
  stat_cor(aes(color = ecotype)) +
      geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey")
Europe_lat
```



```{r}
# Plot without original 21 samples
test <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", c155_global_FvsM %in% c(0,1), samp_length >6) %>%
  ggplot(aes(ecotype, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(size = 3, cex=2.5,alpha = 0.7) +
  geom_boxplot(fill = NA) +
  scale_color_manual(values=c("#d73027","#4575b4")) +
  xlab("Ecotype") +
  ylab("MYH3C3 normalized depth") +
  theme_cowplot(6) +
    theme(legend.position = "top") +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 2.6, size = 3)
test
```

```{r fig.width=5, fig.height=5}
ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", str_detect(samp, "LOBG")) %>%
  ggplot(aes(pop, wg_norm_depth, color = pop)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(size = 3, cex=2.5,alpha = 0.7) +
  geom_boxplot(fill = NA) +
  scale_color_manual(values=c("#d73027","#4575b4")) +
  ggtitle("MYH3C3 read depth") +
  xlab("Ecotype") +
  ylab("Normalized depth of MYH3C3 \n(1x genome coverage)") +
  theme(legend.position = "none",
        text = element_text(size=16)) +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 2.6, size = 5)
```




```{r}
# Make wider version of table to plot correlation between different genes
ga5_roi_cov_wide <- ga5_roi_cov %>% 
  select(desc, samp:type, weighted_mean_depth:sex) %>% 
  pivot_wider(names_from = desc, values_from = wg_norm_depth)
```

```{r}
# Plot pearson correlation between MYH3C3 read depth and duplication region
ga5_roi_cov_wide %>% 
  filter(sex=="female", type=="real") %>%
  ggscatter(x = "MYH3C3", y = "duplication_region",
   add = "reg.line", conf.int = TRUE, alpha=0.5) +
  stat_cor()
```

```{r fig.width=10, fig.height=10}
# world.df = fortify(maps::map("world", plot = FALSE, fill = TRUE))
# 
# worldmap = ggplot() + geom_polygon(data = world.df
#                                    , aes(x = long, y = lat, group = group)
#                                    , fill="gray") +
#                       scale_y_continuous(breaks = (-2:2) * 30) +
#                       scale_x_continuous(breaks = (-4:4) * 45) +
#                       theme_bw()+
#                       theme(axis.line = element_line(colour = "black"),
#                             panel.grid.major = element_blank(),
#                             panel.grid.minor = element_blank(),
#                             panel.border = element_blank(),
#                             panel.background = element_blank()
#                           ) 
# 
# shape = 16
# alpha = 0.7
# size = 5
# 
# worldmap3 = worldmap +
#             geom_point(data = filter(metadata, c155_global_FvsM == 0)
#                        , aes(x = GPS_east, y = GPS_north)
#                        , colour = "#4575b4"
#                        , alpha = alpha
#                        , size = size
#                        , shape = shape) +
#             geom_point(data = filter(metadata, c155_global_FvsM == 1)
#                        , aes(x = GPS_east, y = GPS_north)
#                        , colour = "#d73027"
#                        , alpha = alpha
#                        , size= size
#                        , shape=shape)
# 
# worldmap3+ coord_map("ortho", orientation=c(70,-90,0))
# #            geom_text(data=freshName
# #                      , aes(x=long,y=lat,label=label)
# #                      , colour = "blue") + 
# #            geom_text(data=marineName
# #                      , aes(x=long, y=lat,label=label)
# #                      , colour = "red3")

```

## Final figure plots

```{r}
# Plot MYH3C3 read depth for balanced global selection, excluding low depth original Jones et al. 2012 samples
C3_global <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",
         sex=="female",
         type=="real",
         c155_global_FvsM %in% c(0,1),
         samp_length > 6) %>%
  ggplot(aes(fct_reorder(samp, wg_norm_depth), wg_norm_depth, color = ecotype)) +
    geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_segment( aes(x=samp, xend=samp, y=0, yend=wg_norm_depth)) +
  geom_point(size=1.5) +
  scale_color_manual(values=c("#d73027","#4575b4")) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
   theme_cowplot(6) +
    panel_border(color="black", size=0.75) +
  theme(axis.text.x=element_text(angle = -90, hjust = 0), legend.position = c(0.015, 0.87)) +
  xlab("Individual") +
  ylab("Normalized MYH3C3 depth")
#    ylim(0,3.5)
C3_global
```


```{r}
# Plot with facet
lat <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3",  type=="real", sex=="female", NorthEurope_independent_MvsF_c151 %in% c(0,1)|PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1), samp_length > 6) %>%
  mutate(ocean = case_when(PNW_independent_MvsF_c150 %in% c(0,1)|CaliforniaFreshwater_vs_AllPacificMarine_c153 %in% c(0,1) ~ "Pacific",
                           NorthEurope_independent_MvsF_c151 %in% c(0,1) ~ "Atlantic")) %>% 
  ggplot(aes(GPS_north, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_point(alpha=0.7, size=1.5) +
  geom_smooth(aes(fill = ecotype), method = "lm") +
  stat_cor(aes(color = ecotype), size=2) +
  scale_color_manual(values=c("#d73027","#4575b4")) +
  scale_fill_manual(values=c("#d73027","#4575b4")) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
  facet_wrap(~factor(ocean, levels=c("Pacific", "Atlantic")), scales = "free_x") +
  theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  xlab("Latitude (°N)") +
  ylab("MYH3C3 normalized depth")
  
lat
```

```{r}
hz_plot <- ga5_roi_cov %>% 
  filter(desc == "MYH3C3", sex == "female", !is.na(`used _river_comparisons`), samp_length > 6) %>% 
  mutate(location = as.factor(case_when(acronym == "BIGR" ~ "Big River,\nCalifornia",
                              acronym %in% c("BNMA", "BNST") ~ "Bonsall River,\nBC, Canada",
                              acronym == "LITC" ~ "Little Campbell River,\nBC, Canada",
                              acronym == "TYNE" ~ "Tyne River,\nScotland",
                              acronym == "MIDF" ~ "Midfjardara River,\nIceland")),
         location = fct_relevel(location, "Big River,\nCalifornia", "Little Campbell River,\nBC, Canada")) %>% 
  ggplot(aes(location, wg_norm_depth, color = ecotype)) +
  geom_hline(data = sim_average, aes(yintercept = wg_norm_depth), linetype = "dashed", color = "grey") +
  geom_beeswarm(cex=2, dodge.width = .7, alpha = 0.7) +
  geom_boxplot(fill = NA) +
  scale_color_manual(values=c("#d73027","#4575b4")) +
  scale_y_continuous(limits = c(0,3.7), expand = expansion(mult = c(0, 0))) +
#  ylim(0,2.5) +
  xlab("River") +
  ylab("MYH3C3 normalized depth") +
  theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  stat_compare_means(aes(group = ecotype), label.x = 1, label.y = 3.3, size = 3, label="p.signif")
hz_plot
```



```{r}
# Arrange plots using cowplot
# align all plots vertically
plots <- align_plots(test, C3_global, lat, hz_plot, align = 'v', axis = 'l')

top_row <- plot_grid(
  NULL, plots[[2]],
  labels = c("B", "C"),
  label_size = 12,
  rel_widths = c(1, 3),
  nrow = 1
)

bottom_row <- plot_grid(
  plots[[3]], plots[[4]],
  labels = c("D", "E"),
  rel_widths = c(1, 1),
  label_size = 12,
  nrow = 1
)
final_plot <- plot_grid(top_row, bottom_row, ncol = 1, rel_heights = c(1.2, 1), label_size = 12)
final_plot
```

```{r}
# Save ASE figure to separate file
ggsave(
  "/labs/kingsley/ambenj/myosin_dups/analysis/ecotypic_depth/gasAcuv5_C4masked_nochrY/ecotypic_depth_figure.pdf",
  plot = final_plot,
  scale = 1,
  width = 6.5,
  height = 4.5,
  units = c("in"),
  dpi = 300,
)
```


## PCA of myosin region
```{r}
# # Load vcf file spanning myosin global specific ecopeak
# vcf.fn <- "../ecotypic_depth_gasAcu1-4/227_genomes.final.filtered_chrXIX_2614822-2744076.vcf"
# 
# # Reformat (biallelic version, can also try "copy.num.of.ref")
# snpgdsVCF2GDS(vcf.fn, "test.gds", method="biallelic.only")
# 
# # Summary
# snpgdsSummary("test.gds")
# 
# # Open GDS file
# genofile <- snpgdsOpen("test.gds")
# 
# genofile
```

```{r}
# Run PCA
# pca <- snpgdsPCA(genofile, num.thread=2, autosome.only = FALSE)
# 
# snpgdsClose(genofile)
# 
# # variance proportion (%)
# pc.percent <- pca$varprop*100
# head(round(pc.percent, 2))
```

```{r}
# make a data.frame
# tab <- data.frame(sample.id = pca$sample.id,
#     EV1 = pca$eigenvect[,1],    # the first eigenvector
#     EV2 = pca$eigenvect[,2],    # the second eigenvector
#     EV3 = pca$eigenvect[,3],    # the third eigenvector
#     EV4 = pca$eigenvect[,4],    # the fourth eigenvector
#     stringsAsFactors = FALSE)
# 
# tab_annot <- tab %>% 
#   mutate(samp = str_replace_all(sample.id, '\\|', '_'),
#          samp = str_replace_all(samp, '\\#', '_')) %>% 
#   left_join(., filter(roi_cov, desc == "MYH3C3"), by=c("samp" = "samp"))
# tab_annot

```

```{r}
# p <- tab_annot %>% 
#   ggplot(aes(EV1, EV2, color = wg_norm_depth)) +
#   geom_point(aes(shape = ecotype), size = 2,) +
#   scale_color_viridis_c(option = "magma") 
# p
```
```{r fig.width=20, fig.height=15}
# p +
#   geom_label_repel(aes(label = acronym),
#                    size = 3,
#                   box.padding   = 0.1, 
#                   point.padding = 0.1,
#                   segment.color = 'grey50')
```

```{r fig.width=20, fig.height=15}
# p +
#   geom_label_repel(aes(label = samp),
#                    size = 3,
#                   box.padding   = 0.1, 
#                   point.padding = 0.1,
#                   segment.color = 'grey50')
```

```{r fig.width=15, fig.height=15}
# Marine cluster zoom in
# p +
#   xlim(.1, .15) +
#   ylim(-.045, -.025) +
#   geom_label_repel(aes(label = samp),
#                    size = 3,
#                   box.padding   = 0.1, 
#                   point.padding = 0.1,
#                   segment.color = 'grey50')
```
```{r fig.width=20, fig.height=10}
# Bottom marine cluster samples
# tab_annot %>% 
#   filter(between(EV1, .1, .15), between(EV2, -.045, -.025)) %>% 
#   arrange(-wg_norm_depth) %>% 
#   select(samp, pop, water_type, ecotype, wg_norm_depth)
```

```{r fig.width=20, fig.height=20}
# p +
#   xlim(-.07, -.01) +
#   ylim(-.03, 0) +
#   geom_label_repel(aes(label = samp),
#                    size = 3,
#                   box.padding   = 0.1, 
#                   point.padding = 0.1,
#                   segment.color = 'grey50')
```

```{r fig.width=20, fig.height=20}
# p +
#   xlim(-.06, -.045) +
#   ylim(-.0275, -.02) +
#   geom_label_repel(aes(label = samp),
#                    size = 3,
#                   box.padding   = 0.1, 
#                   point.padding = 0.1,
#                   segment.color = 'grey50')
```

```{r fig.width=20, fig.height=10}
# Bottom freshwater cluster samples
# tab_annot %>% 
#   filter(between(EV1, -.06, -.04), between(EV2, -.03, -.02)) %>% 
#   arrange(wg_norm_depth) %>% 
#   select(samp, pop, water_type, ecotype, wg_norm_depth, GPS_north, GPS_east)
```

```{r fig.width=20, fig.height=10}
# Top cluster samples
# tab_annot %>% 
#   filter(EV2 > 0.04) %>% 
#   arrange(wg_norm_depth) %>% 
#   select(samp, pop, water_type, ecotype, wg_norm_depth)
```
