---
title: "Myosin coverage - Transplant pop pool-seq with mapq3 filter"
author: "Alyssa Yoxsimer"
date: "5/10/2024"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse); theme_set(theme_bw())
library(fs);
library(cowplot); theme_set(theme_cowplot(font_size=6));
library(trend)
```

```{r}
# File paths
# roi_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/06_samtools_coverage_roi/roi_coverage_mapq3.txt"
# wg_path <-"/labs/kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/06_samtools_coverage_wg"
# af_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/transplant_pops/Veeramah_PoolSeq_AFs_corrected.filtered_chrXIX.txt"
# out_file <- "/labs/kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/poolseq_depth_figure.pdf"
```

```{r}
# File paths
roi_file <- "/Volumes/lab_kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/06_samtools_coverage_roi/roi_coverage_mapq3.txt"
wg_path <-"/Volumes/lab_kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/06_samtools_coverage_wg"
af_file <- "/Volumes/lab_kingsley/ambenj/myosin_dups/analysis/transplant_pops/Veeramah_PoolSeq_AFs_corrected.filtered_chrXIX.txt"
out_file <- "/Volumes/lab_kingsley/ambenj/myosin_dups/analysis/transplant_pops/gasAcuv5_C4masked/poolseq_depth_figure.pdf"
```

## Depth analysis
```{r}
# Read roi file
roi <- read_tsv(roi_file) %>% 
  mutate(bam = str_remove_all(bam, "05_mkdup_index/|.PE_ME_merged.sort.Mkdup.realign.BQrecal.realignGA5_C4masked.sort.merged.mkdup.bam|.PE_ME_merged.sort.realign.BQrecal.realignGA5_C4masked.sort.merged.mkdup.bam|.PE_ME_merged.sort.Mkdup.rename.realign.BQrecal.realignGA5_C4masked.sort.merged.mkdup.bam")) %>% 
  rename(samp = bam)
 # separate(bam, into = c("pop", "year"), sep="-", convert = TRUE)
roi

```

```{r message=FALSE}
# Read whole genome files
files <- dir_ls(path = wg_path, glob = "*wg_coverage.txt")

# function to add file name to dataframe
read_and_record_filename <- function(filename){read_tsv(filename) %>%
  mutate(filename = path_file(filename))
  }

# gather files into one dataframe
wg <- map_df(files, read_and_record_filename) %>% 
  separate(filename, into=c("samp"), sep="\\.", extra="drop")
wg
```
```{r fig.width=10, fig.height=5}
wg %>% 
  filter(!chr %in% c("chrM", "chrUn")) %>% 
  ggplot(aes(chr, meandepth)) +
  geom_point() +
  coord_flip() +
  facet_wrap(~samp, nrow = 3)
```


```{r}
# get whole genome mean depth (excluding chrUn, chrM, chrXIX, chrY)
wg_cov <- wg %>% 
  filter(!chr %in% c("chrUn", "chrM", "chrXIX", "chrY")) %>% 
  group_by(samp) %>% 
  summarize(weighted_mean_depth = weighted.mean(meandepth, endpos))
wg_cov

# Does it look like even depth ratios? Would expect chrY / chrXIX depth = 0.5   3x for every Y with 50/50 so chrXIX / chrY depth of 3 -> might later years more female skewed...can I correct for this?
sex_ratio <- wg %>% 
  filter(chr %in% c("chrXIX", "chrY")) %>% 
  pivot_wider(id_cols = samp, names_from = chr, values_from = meandepth) %>% 
  mutate(sex_ratio= chrXIX / chrY)
sex_ratio
```

```{r}
coverage <- left_join(wg_cov, sex_ratio)
coverage
```

```{r fig.width=12}
# Add additional depth info to rois
roi_cov <- left_join(roi, coverage) %>% 
  separate(samp, into = c("pop", "year"), sep="-", convert = TRUE) %>%
  mutate(wg_norm_depth = meandepth / weighted_mean_depth,
          chrXIX_norm_depth = meandepth / chrXIX,
          Y_norm_depth = meandepth / chrY,
         yrs_since_found = case_when(pop == "CH" ~ year - 2009,
                                     pop == "SC" ~ year - 2011,
                                     pop == "RS" ~ year - 2009,
                                     pop == "LB" ~ year - 1985),
         pop = case_when(pop == "RS" ~ "RABS",
                         TRUE ~ pop),
         pop = fct_relevel(pop, c("RABS", "SC", "CH", "LB")),
         desc = as.factor(desc),
         desc = fct_relevel(desc, "NLRC5", "SYT19", "CALB2A", "HTRA1A", "MYH3C1", "MYH3C2", "MYH3C3", "MYH3C1y", "MYH3C2y", "MYH3C3y"))

# Is it ok to use all of chrXIX for normalization?
roi_cov %>% 
  filter(pop == "CH" & year == "2017" & str_detect(desc, "MYH")) %>% 
  summarise(sum = sum(numreads),
            sum_covbase = sum(covbases))
wg %>% 
  filter(samp == "CH-2017" & chr == "chrXIX")

roi_cov
```
If the sex ratios are 1F:1M: 3:1 chromosome ratio (chrXIX:chrY)
If the sex ratios are 2F:1M: 5:1 chromosome ratio
If the sex ratios are 1F:2M: 1:2 chromosome ratio
```{r}
roi_cov %>% 
  filter(desc == "HTRA1A") %>% 
  ggplot(aes(yrs_since_found, sex_ratio, color = pop)) +
  geom_hline(yintercept = 3, linetype = "dashed") +
  geom_line(linewidth = 1.5) +
  geom_point(size = 3) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Sex chromosome ratio (chrXIX / chrY depth)")
```
```{r}
roi_cov %>% 
  filter(desc == "HTRA1A") %>% 
  mutate(chrXIX = chrXIX / weighted_mean_depth,
         chrY = chrY / weighted_mean_depth) %>% 
  pivot_longer(cols = chrXIX:chrY, names_to = "chromosome", values_to="depth_1x") %>% 
  ggplot(aes(yrs_since_found, depth_1x, color = pop, group = interaction(chromosome, pop))) +
  geom_line(aes(linetype = chromosome), linewidth = 1.5) +
  geom_point(size = 3) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  ggtitle("Sex chromsome depths") +
  xlab("Years since founding") +
  ylab("Normalized depth (1x genome coverage)")
```


```{r}
roi_cov %>% 
  filter(chr!= "chrY") %>% 
  ggplot(aes(yrs_since_found, wg_norm_depth, color = pop)) +
  geom_line(size = 1) +
  geom_point(size = 1) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Normalized depth (1x genome coverage)") +
  facet_wrap(~desc, nrow = 2)
```

```{r}
roi_cov %>% 
  filter(desc %in% c("NLRC5", "STY19","CALB2A", "HTRA1A", "MYH3C1", "MYH3C2", "MYH3C3")) %>% 
  ggplot(aes(yrs_since_found, wg_norm_depth, color = pop)) +
  geom_line(linewidth = 1) +
  geom_point(size = 1.5) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Normalized depth (1x genome coverage)") +
  facet_wrap(~desc, nrow = 2)
```

```{r}
roi_cov %>% 
  filter(chr!= "chrY") %>% 
  ggplot(aes(yrs_since_found, chrXIX_norm_depth, color = pop)) +
  geom_line(size = 1) +
  geom_point(size = 1.5) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Normalized depth (1x chrXIX coverage)") +
#  ylim(0,1.7) +
  facet_wrap(~desc, nrow = 2)
```


```{r}
roi_cov %>% 
  filter(desc == "MYH3C3") %>% 
  ggplot(aes(yrs_since_found, chrXIX_norm_depth, color = pop)) +
  geom_line(size = 1) +
  geom_point(size = 1.5) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Normalized depth (1x chrXIX coverage)")
```


```{r}
roi_cov %>% 
  filter(chr== "chrY") %>% 
  ggplot(aes(yrs_since_found, Y_norm_depth, color = pop)) +
  geom_line(size = 1) +
  geom_point(size = 1.5) +
#  scale_color_manual(values=c("#9970ab", "#5aae61")) +
  xlab("Years since founding") +
  ylab("Normalized depth (1x chrY coverage)") +
#  ylim(0,1.7) +
  facet_wrap(~desc, nrow = 1)
```
## Variant analysis
```{r}
af <- read_tsv(af_file) %>% 
  filter(pos < 2900000, pos > 2500000, `RS-2009` > 0.9) %>% 
  pivot_longer("RS-2009":"LB-2017", names_to = "pop", values_to = "af") %>% 
  separate(pop, sep = "-", into = c("pop", "year"), convert = TRUE) %>% 
  mutate(yrs_since_found = case_when(pop == "CH" ~ year - 2009,
                                     pop == "SC" ~ year - 2011,
                                     pop == "LB" ~ year - 1985,
                                     pop == "RS" ~ year - 2009),
         pop = case_when(pop == "RS" ~ "RABS",
                         TRUE ~ pop),
         pop = fct_relevel(pop, c("RABS", "SC", "CH", "LB")),
         ref_af = 1-af)
af
```

```{r}
af %>% 
  filter(pos > 2673873, pos > 2674873, pop!="RABS") %>% 
  ggplot(aes(yrs_since_found, ref_af, color = pop, fill=pop)) +
  stat_summary(fun.data = "mean_cl_boot", geom = "smooth", se = TRUE) +
#  geom_point(data = filter(af, pop == "RABS"), size = 2.5, shape=5, alpha=0.1) +
#  geom_smooth(method="lm") 
#  geom_line(data = filter(af, pop != "RABS"),size = 0.1, alpha=0.1) +
    theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = c(0.8, 0.3)) +
  scale_color_manual("Population", values=c("#cc79a7", "#7a669b", "#56b4e9")) +
  scale_fill_manual("Population", values=c("#cc79a7", "#7a669b", "#56b4e9")) +
  xlab("Years since founding") +
  ylab("FW Allele frequency")
```
```{r}
af %>% 
  filter(pos > 2673873, pos > 2674873, pop!="RABS") %>% 
  ggplot(aes(yrs_since_found, ref_af, color = pop, group = interaction(pos, pop))) +
#  geom_point(data = filter(af, pop == "RABS"), size = 2.5, shape=5, alpha=0.1) +
  geom_line(data = filter(af, pop != "RABS"),size = 0.025, alpha=0.8) +
  stat_summary(
    fun = median,
    geom = 'line',
    aes(group=pop),
    size=3,
    line_type = "dashed") +
    theme_cowplot(8) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  scale_color_manual("Population", values=c("#cc79a7", "#7a669b", "#56b4e9")) +
  scale_y_continuous(limits = c(0,1), expand = expansion(mult = c(0, 0))) +
  scale_x_continuous(limits = c(-.2,32.2), expand = expansion(mult = c(0, 0))) +
  xlab("Years since founding") +
  ylab("FW Allele frequency")
```


## Final plots
```{r}
depth_plot <- roi_cov %>% 
  filter(desc == "MYH3C3") %>% 
  ggplot(aes(yrs_since_found, chrXIX_norm_depth, color = pop)) +
  geom_point(data = filter(roi_cov, desc == "MYH3C3", pop == "RABS"), size = 2.5, shape=5, stroke=1.5) +
  geom_line(data = filter(roi_cov, desc == "MYH3C3", pop != "RABS"),size = 1) +
  geom_point(data = filter(roi_cov, desc == "MYH3C3", pop != "RABS"), size = 1.5) +
    theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  scale_color_manual("Population", values=c("#d73027", "#cc79a7", "#7a669b", "#56b4e9")) +
    scale_y_continuous(limits = c(0,2.2), expand = expansion(mult = c(0, 0))) +
  scale_x_continuous(limits = c(-.5,32.5), expand = expansion(mult = c(0, 0))) +
  xlab("Years since founding") +
  ylab("MYH3C3 normalized depth")
depth_plot
```
```{r}
# Mann-Kendall test for CH read depth
mk_CH <- mk.test(pull(filter(roi_cov, pop == "CH", desc=="MYH3C3"), chrXIX_norm_depth), alternative = "greater" )
mk_CH
```
```{r}
# Mann-Kendall test for SC read depth
mk_SC <- mk.test(pull(filter(roi_cov, pop == "SC", desc=="MYH3C3"), chrXIX_norm_depth), alternative = "greater")
mk_SC
mk_SC$p.value
```
```{r}
# Mann-Kendall test for LB read depth
mk_LB <- mk.test(pull(filter(roi_cov, pop == "LB", desc=="MYH3C3"), chrXIX_norm_depth), alternative = "greater" )
mk_LB
```



```{r}
# Variant frequency plot
af_plot <- af %>% 
  ggplot(aes(yrs_since_found, ref_af, color = pop)) +
  geom_point(data = filter(af, pos == 2744769, pop == "RABS"), size = 2.5, shape=5, stroke=1.5) +
  geom_line(data = filter(af, pos == 2744769, pop != "RABS"),size = 1) +
  geom_point(data = filter(af, pos == 2744769, pop != "RABS"), size = 1.5) +
    theme_cowplot(6) +
  panel_border(color="black", size=0.75) +
  theme(legend.position = "none") +
  scale_color_manual("Population", values=c("#d73027", "#cc79a7", "#7a669b", "#56b4e9")) +
    scale_y_continuous(limits = c(0,1), expand = expansion(mult = c(0, 0))) +
  scale_x_continuous(limits = c(-.5,32.5), expand = expansion(mult = c(0, 0))) +
  xlab("Years since founding") +
  ylab("FW allele frequency")
```
```{r}
final_plot1 <- plot_grid(depth_plot, af_plot, ncol = 1, rel_heights = c(1, 1), labels = c("B","C"), label_size = 12)
final_plot1
```
```{r}
final_plot2 <- plot_grid(NULL, depth_plot, nrow = 1, rel_widths = c(1, 1), labels = c("E","F"), label_size = 12)
final_plot2
```


```{r}
ggsave(
  out_file,
  plot = final_plot2,
  scale = 1,
  width = 6.5,
  height = 1.75,
  units = c("in"),
  dpi = 300,
)
```

```{r}
final_plot_h <- plot_grid(depth_plot, af_plot, nrow = 1, rel_widths = c(1, 1), labels = c("B","C"), label_size = 12)
final_plot_h
```

